recipes/volume_extend.rb

```diff
@@ -0,0 +1,44 @@
+node['gluster']['server']['volumes'].each do |volume_name, volume_values|
+  # 1. Get the current size of the logical volume
+  # 2. Compare to the size set for the gluster volume
+  # 3. If different, run a resize action against that volume
+  # ToDO: change hardcoded VG name gluster into an attribute
+  require 'lvm'
+
+  LVM::LVM.new do |lvm|
+    lvm.logical_volumes.each do |lv|
+      # I'm ignoring these as I think this layout helps readability
+      # rubocop:disable Style/Next, Style/CaseIndentation, Lint/EndAlignment
+      if lv.full_name.to_s == "gluster/#{volume_name}"
+        lv_size_cur = lv.size.to_i
+        # Borrowed from the lvm cookbook
+        volume_lv_size_req = case volume_values['size']
+          when /^(\d+)(k|K)$/
+            (Regexp.last_match[1].to_i * 1024)
+          when /^(\d+)(m|M)$/
+            (Regexp.last_match[1].to_i * 1_048_576)
+          when /^(\d+)(g|G)$/
+            (Regexp.last_match[1].to_i * 1_073_741_824)
+          when /^(\d+)(t|T)$/
+            (Regexp.last_match[1].to_i * 1_099_511_627_776)
+          when /^(\d+)$/
+            Regexp.last_match[1].to_i
+          else
+            Chef::Application.fatal!("Invalid size #{Regexp.last_match[1]} for lvm resize", 2)
```

Comment:
Copied from the LVM cookbook so it's what they used. It does make sense to re-use the code though, an exit code 2 suggesting that the issue is with LVM. Would you prefer it to be exit 1?